airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/RefreshSchemaActivityImpl.java

```diff
@@ -8,52 +8,72 @@
 
 import datadog.trace.api.Trace;
 import io.airbyte.api.client.generated.SourceApi;
-import io.airbyte.api.client.invoker.generated.ApiException;
 import io.airbyte.api.client.model.generated.SourceDiscoverSchemaRequestBody;
+import io.airbyte.commons.features.EnvVariableFeatureFlags;
 import io.airbyte.config.ActorCatalogFetchEvent;
 import io.airbyte.config.persistence.ConfigRepository;
+import jakarta.inject.Singleton;
 import java.io.IOException;
 import java.time.OffsetDateTime;
 import java.util.Optional;
 import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
+@Singleton
 public class RefreshSchemaActivityImpl implements RefreshSchemaActivity {
 
   private final Optional<ConfigRepository> configRepository;
 
   private final SourceApi sourceApi;
+  private final EnvVariableFeatureFlags envVariableFeatureFlags;
 
-  public RefreshSchemaActivityImpl(Optional<ConfigRepository> configRepository, SourceApi sourceApi) {
+  public RefreshSchemaActivityImpl(Optional<ConfigRepository> configRepository,
+                                   SourceApi sourceApi,
+                                   EnvVariableFeatureFlags envVariableFeatureFlags) {
     this.configRepository = configRepository;
     this.sourceApi = sourceApi;
+    this.envVariableFeatureFlags = envVariableFeatureFlags;
   }
 
   @Override
   @Trace(operationName = ACTIVITY_TRACE_OPERATION_NAME)
-  public boolean shouldRefreshSchema(UUID sourceCatalogId) throws IOException {
+  public boolean shouldRefreshSchema(UUID sourceCatalogId) {
     // if job persistence is unavailable, default to skipping the schema refresh
-    if (configRepository.isEmpty()) {
+    if (configRepository.isEmpty() || !envVariableFeatureFlags.autoDetectSchema()) {
       return false;
     }
 
     return !schemaRefreshRanRecently(sourceCatalogId);
   }
 
   @Override
-  public void refreshSchema(UUID sourceCatalogId) throws ApiException {
-    SourceDiscoverSchemaRequestBody requestBody =
-        new SourceDiscoverSchemaRequestBody().sourceId(sourceCatalogId).disableCache(true);
-    sourceApi.discoverSchemaForSource(requestBody);
-  }
+  public void refreshSchema(UUID sourceCatalogId, UUID connectionId) {
+    if (!envVariableFeatureFlags.autoDetectSchema()) {
+      return;
+    }
 
-  private boolean schemaRefreshRanRecently(UUID sourceCatalogId) throws IOException {
-    Optional<ActorCatalogFetchEvent> mostRecentFetchEvent = configRepository.get().getMostRecentActorCatalogFetchEventForSource(sourceCatalogId);
+    SourceDiscoverSchemaRequestBody requestBody =
+        new SourceDiscoverSchemaRequestBody().sourceId(sourceCatalogId).disableCache(true).connectionId(connectionId);
 
-    if (mostRecentFetchEvent.isEmpty()) {
-      return false;
+    try {
+      sourceApi.discoverSchemaForSource(requestBody);
+    } catch (final Exception e) {
+      log.error("Attempted schema refresh, but failed with error: ", e);
```

Comment:
nit: Could be worth adding a comment explaining that we swallow the exception here to avoid blocking the replication if we fail to refresh.